language/libra-tools/writeset-transaction-generator/src/writeset_builder.rs

```diff
@@ -0,0 +1,199 @@
+            vec![],
+            vec![
+                Value::transaction_argument_signer_reference(libra_root_address()),
+                Value::u64(version),
+            ],
+        )
+    }
+}
+
+/// Convert the transaction arguments into Move values.
+fn convert_txn_args(args: &[TransactionArgument]) -> Vec<Value> {
+    args.iter()
+        .map(|arg| match arg {
+            TransactionArgument::U8(i) => Value::u8(*i),
+            TransactionArgument::U64(i) => Value::u64(*i),
+            TransactionArgument::U128(i) => Value::u128(*i),
+            TransactionArgument::Address(a) => Value::address(*a),
+            TransactionArgument::Bool(b) => Value::bool(*b),
+            TransactionArgument::U8Vector(v) => Value::vector_u8(v.clone()),
+        })
+        .collect()
+}
+
+fn move_module_changes(modules: &[CompiledModule]) -> MoveChanges {
+    let mut shadow_changeset = MoveChanges::new();
+    for module in modules.iter() {
+        let id = module.self_id();
+        let mut module_bytes = vec![];
+        module.serialize(&mut module_bytes).unwrap();
+        shadow_changeset.publish_or_overwrite_module(id, module_bytes);
+    }
+    shadow_changeset
+}
+
+fn shadow_reconfiguration<S: StateView>(state_view: &S, move_changes: &mut MoveChanges) {
+    let move_vm = MoveVM::new();
+    let state_view_storage = RemoteStorage::new(state_view);
+    let mut session = move_vm.new_session(&state_view_storage);
+    session
+        .execute_function(
+            &ModuleId::new(
+                account_config::CORE_CODE_ADDRESS,
+                Identifier::new("LibraConfig").unwrap(),
+            ),
+            &Identifier::new("reconfigure_").unwrap(),
+            vec![],
+            vec![],
+            libra_root_address(),
+            &mut CostStrategy::system(&ZERO_COST_SCHEDULE, GasUnits::new(100_000_000)),
+            &NoContextLog::new(),
+        )
+        .unwrap();
+
+    let effect = session.finish().unwrap();
+    for (account, ops) in effect.resources {
+        for (struct_tag, op) in ops {
+            match op {
+                Some((layout, value)) => {
+                    let blob = value.simple_serialize(&layout).unwrap();
+                    move_changes
+                        .publish_resource(account, struct_tag, blob)
+                        .unwrap();
+                }
+                None => move_changes
+                    .unpublish_resource(account, struct_tag)
+                    .unwrap(),
+            }
+        }
+    }
+}
+
+pub fn build_changeset<S: StateView, F>(
+    state_view: &S,
+    override_move_changes: Option<MoveChanges>,
+    procedure: F,
+    to_be_published_modules: &[CompiledModule],
+) -> ChangeSet
+where
+    F: FnOnce(&mut GenesisSession<DeltaStorage<RemoteStorage<S>>>),
+{
+    let move_vm = MoveVM::new();
+    let mut move_changes = override_move_changes.unwrap_or_else(MoveChanges::new);
+    shadow_reconfiguration(state_view, &mut move_changes);
```

Comment:
A reconfiguration is executed in shadow mode so that no real reconfiguration will be triggered during the execution phase.